File: java/sandbox_project/src/AddressTest.java
import junit.framework.TestCase;

public class AddressTest extends TestCase {
	
	public void testBuilderDefaults() {
		Address expected = new Address.Builder("Chicago", "IL").build();
		assertEquals("State is IL", "IL", expected.getState());
		assertEquals("City is Chicago", "Chicago", expected.getCity());
	}
	
	public void testBuilderStreetStateZip() {
		Address expected = new Address.Builder("Chicago", "IL").street("this is an address").zip("11111").build();
		assertEquals("Address is this is an address", "this is an address", expected.getStreet());
		assertEquals("Zip is 11111", "11111", expected.getZip());
		assertEquals("State is IL", "IL", expected.getState());
		assertEquals("City is Chicago", "Chicago", expected.getCity());
	}

	// Try a different order
	public void testBuilderZipStreetState() {
		Address expected = new Address.Builder("Chicago", "IL").zip("11111").street("this is an address").build();
		assertEquals("Address is this is an address", "this is an address", expected.getStreet());
		assertEquals("Zip is 11111", "11111", expected.getZip());
		assertEquals("State is IL", "IL", expected.getState());
		assertEquals("City is Chicago", "Chicago", expected.getCity());	
	}

}

File: ruby/trying_builder_pattern/address_test.rb
require "address"
require "test/unit"

class AddressTest < Test::Unit::TestCase
  
  def testDefaultValues
    myaddress = Address.new
    assert_equal("Chicago", myaddress.city)
    assert_equal("IL", myaddress.state)
  end
  
  def testStreetStateZip 
    myaddress = Address.new do 
      self.street = "this is the street"
      self.zip = "11111"
    end
    assert_equal("Chicago", myaddress.city)
    assert_equal("IL", myaddress.state)
    assert_equal("this is the street", myaddress.street)
    assert_equal("11111", myaddress.zip)
  end

  def testZipStreetState
    myaddress = Address.new do 
      self.zip = "11111"
      self.street = "this is the street"
    end
    assert_equal("this is the street", myaddress.street) 
    assert_equal("11111", myaddress.zip)
  end


end